src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -395,6 +395,13 @@ private void StartWriting(byte[] buffer, int offset, int count, AsyncProtocolReq
 
                 do
                 {
+                    if (count == 0 && !SslStreamPal.CanEncryptEmptyMessage)
```

Comment:
I think the Windows behavior allows for a zero-byte keepalive message to be sent; and that functionality would go away.  IE6 was known to crash on reading 0-payload-byte TLS messages, though; so I don't know that anyone really uses that behavior.

The fact that this was in a do/while loop suggests that the original authors wanted to enable that scenario, so it'd really require that Cristian or David chime in.

I agree that changing it to a pure while is an easier change; but I was aiming for "least impactful".